composables/zksync/deposit/useTransaction.ts

```diff
@@ -1,23 +1,131 @@
+import { checksumAddress } from "viem";
+import { mainnet, sepolia } from "viem/chains";
+import { BrowserProvider, Contract, L1Signer, Provider, types } from "zksync-ethers";
+import { getERC20DefaultBridgeData } from "zksync-ethers/build/utils";
+
 import { useSentryLogger } from "@/composables/useSentryLogger";
+import { L1_BRIDGE_ABI } from "@/data/abis/l1BridgeAbi";
+
+import { useViemClient } from "./useViemClient";
 
 import type { DepositFeeValues } from "@/composables/zksync/deposit/useFee";
 import type { BigNumberish } from "ethers";
-import type { L1Signer } from "zksync-ethers";
 
 export default (getL1Signer: () => Promise<L1Signer | undefined>) => {
   const status = ref<"not-started" | "processing" | "waiting-for-signature" | "done">("not-started");
   const error = ref<Error | undefined>();
   const ethTransactionHash = ref<string | undefined>();
   const eraWalletStore = useZkSyncWalletStore();
+  const { eraNetwork } = useZkSyncProviderStore();
   const { captureException } = useSentryLogger();
+  const chain = eraNetwork.key === "mainnet" ? mainnet : sepolia;
```

Comment:
this doesn't make a lot of sense to me.. Why not use l1 network config of selected network?